pkg/archive/archive.go

```diff
@@ -1501,3 +1502,19 @@ func (archive *TempArchive) Read(data []byte) (int, error) {
 	}
 	return n, err
 }
+
+// UncompressedDigest returns the digest of the uncompressed tar stream.
+func UncompressedDigest(alg digest.Algorithm, compressedTar io.Reader) (digest.Digest, error) {
+	rd, err := DecompressStream(compressedTar)
+	if err != nil {
+		return "", err
+	}
+
+	defer rd.Close()
+
+	digester := alg.Digester()
```

Comment:
Also note that this will panic if `"crypto/sha256"` is not imported (which is currently not by default, see https://github.com/opencontainers/go-digest/pull/64)

https://github.com/moby/moby/blob/794f7127efd841e023dbd88051c577489f6cd3cf/vendor/github.com/opencontainers/go-digest/algorithm.go#L112

https://github.com/moby/moby/blob/794f7127efd841e023dbd88051c577489f6cd3cf/vendor/github.com/opencontainers/go-digest/algorithm.go#L116-L136

Reply:
Oh, TIL, thanks!